larva/src/main/java/org/frankframework/larva/LarvaTool.java

```diff
@@ -1518,124 +1524,122 @@ public static Message getQueryFromSender(FixedQuerySender sender) {
+			} else if ("org.frankframework.larva.XsltProviderListener".equals(properties.get(queueName + QueueCreator.CLASS_NAME_PROPERTY_SUFFIX))) {
+				stepPassed = executeQueueRead(step, stepDisplayName, properties, queues, queueName, fileName, fileContent);  // XsltProviderListener has .read and .write reversed
+			} else {
+				stepPassed = executeQueueWrite(stepDisplayName, queues, queueName, fileContent, correlationId, null);
 			}
 		}
 
 		return stepPassed;
 	}
 
-	public String readFile(String fileName) {
-		String result = null;
-		String encoding = null;
+	public Message readFile(@Nonnull String fileName) {
+		String encoding;
 		if (fileName.endsWith(".xml") || fileName.endsWith(".wsdl")) {
-			// Determine the encoding the XML way but don't use an XML parser to
-			// read the file and transform it to a string to prevent changes in
-			// formatting and prevent adding a xml declaration where this is
-			// not present in the file. For example, when using a
-			// WebServiceSender to send a message to a WebServiceListener the
-			// xml message must not contain a xml declaration.
-			try (InputStream in = new FileInputStream(fileName)) {
-				XMLInputFactory factory = XMLInputFactory.newInstance();
-				factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
-				factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);
-
-				XMLStreamReader parser = factory.createXMLStreamReader(in);
-				encoding = parser.getEncoding();
-				parser.close();
-			} catch (IOException | XMLStreamException e) {
-				errorMessage("Could not determine encoding for file '" + fileName + "': " + e.getMessage(), e);
-			}
+			encoding = parseEncodingFromXml(fileName);
 		} else if (fileName.endsWith(".utf8") || fileName.endsWith(".json")) {
 			encoding = "UTF-8";
-		} else {
+		} else if (fileName.endsWith(".ISO-8859-1")) {
 			encoding = "ISO-8859-1";
+		} else {
+			encoding = null;
 		}
-		if (encoding != null) {
-			Reader inputStreamReader = null;
+		Message result = new FileMessage(new File(fileName), encoding);
+		if (encoding == null) {
 			try {
-				StringBuilder stringBuilder = new StringBuilder();
-				inputStreamReader = StreamUtil.getCharsetDetectingInputStreamReader(new FileInputStream(fileName), encoding);
-				char[] cbuf = new char[4096];
-				int len = inputStreamReader.read(cbuf);
-				while (len != -1) {
-					stringBuilder.append(cbuf, 0, len);
-					len = inputStreamReader.read(cbuf);
-				}
-				result = stringBuilder.toString();
-			} catch (Exception e) {
-				errorMessage("Could not read file '" + fileName + "': " + e.getMessage(), e);
-			} finally {
-				if (inputStreamReader != null) {
-					try {
-						inputStreamReader.close();
-					} catch(Exception e) {
-						errorMessage("Could not close file '" + fileName + "': " + e.getMessage(), e);
-					}
+				Charset charset = MessageUtils.computeDecodingCharset(result);
```

Comment:
Volgens mij kan je de charset ook op `AUTO` zetten. Dan doet hij dit intern in Message als en wanneer het nodig is.

Reply:
Ik zal eens kijken of dat werkt of dat dat wat kapot maakt.

Reply:
Helaas werkt charset `AUTO` niet. Er wordt `auto` meegegeven aan de `SerializableFileReference` en die doet er nog niks mee. Als gevolg gaan er heel veel tests mis.
De encoding `null` laten werkt dan beter, voorlopig.